### ISSUE TITLE
Vue: Unknown custom element error on stories.mdx stories since upgrading to sb@6.1.11^

### ISSUE THREAD

**Issue by atflick** [storybookjs/storybook#13697]
**Describe the bug**
*.stories.mdx isn't properly loading custom components since Storybook version 6.1.11^.  I have a styleguide portion of my Storybook where I am using mdx stories.  I have wrote some custom Vue components that I reuse that has some example stuff and a part that shows code markup using prismjs.  Ever since upgrading from 6.1.10 I get errors in the console:

 `[Vue warn]: Unknown custom element: <DisplayBlock> - did you register the component correctly? For recursive components, make sure to provide the "name" option.`

Webpack compiles fine without errors, these errors show in the browser and my components aren't displayed. I think something changed since 6.1.10 that affected custom code or I am just not or have not been doing something correctly.  

**To Reproduce**

1. Create Vue Storybook project
2. Add files below
3. Run storybook
4. Go to Typography Styleguide page
5. See that components aren't displaying

**Expected behavior**
My components should be rendering properly

**Screenshots**

![image](https://user-images.githubusercontent.com/18269370/105379039-704c9f80-5bda-11eb-9571-fccfcbb46666.png)

**Code snippets**
I simplified these a little bit to reproduce more easily...

My custom component `DisplayBlock.vue`

```vue
<template>
  <div class="display-block">
    <h2 class="sg-section-title">{{ title }}</h2>
    <slot name="description">
      <p v-if="description">{{ description }}</p>
    </slot>
    <div class="display-block-example">
      <div class="display-block-example-top">
        <div class="display-block-example-inner">
          <slot name="example"></slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script>

export default {
  name: 'DisplayBlock',
  props: {
    title: String,
    description: String,
    language: {
      type: String,
      default: 'markup'
    }
  }
}
</script>
```

My `Paragraphs.vue` file which uses `DisplayBlock` is imported to my `Typography.stories.mdx` file
```vue
<template>
  <div class="paragraphs">
    <DisplayBlock title="Paragraphs">
      <template v-slot:example>
        <div v-html="paragraph"></div>
      </template>
    </DisplayBlock>
  </div>
</template>

<script>
import DisplayBlock from '@components/DisplayBlock';

  export default {
    name: 'Paragraphs',
    components: {
      DisplayBlock
    },
    data() {
      return {
        paragraph:'<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.</p>'
      }
    }
  }
</script>
```

My `Typography.stories.mdx` file
```mdx
import Paragraphs from '@components/Styleguide/Typography/Paragraphs.vue'
import { Meta, Canvas, Story, Preview, DocsPage } from '@storybook/addon-docs/blocks'
<Meta title="Styleguide/Typography" />

  # Typography
  ---

<Story name='Paragraphs'>
  {{
    components: { Paragraphs },
    template: '<Paragraphs />'
  }}
</Story>
```

**System**
Everything works with this:
```
Environment Info:

  System:
    OS: Linux 4.4 Ubuntu 20.04.1 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
  Binaries:
    Node: 12.19.0 - ~/.nvm/versions/node/v12.19.0/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v12.19.0/bin/npm
  npmPackages:
    @storybook/addon-a11y: 6.1.10 => 6.1.10
    @storybook/addon-actions: 6.1.10 => 6.1.10
    @storybook/addon-backgrounds: 6.1.10 => 6.1.10
    @storybook/addon-centered: 5.3.21 => 5.3.21
    @storybook/addon-contexts: 5.3.21 => 5.3.21
    @storybook/addon-docs: ^6.1.10 => 6.1.10
    @storybook/addon-essentials: ^6.1.10 => 6.1.10
    @storybook/addon-knobs: 6.1.10 => 6.1.10
    @storybook/addon-links: 6.1.10 => 6.1.10
    @storybook/addon-notes: 5.3.21 => 5.3.21
    @storybook/addon-options: 5.3.21 => 5.3.21
    @storybook/addon-storyshots: 6.1.10 => 6.1.10
    @storybook/addon-storysource: 6.1.10 => 6.1.10
    @storybook/addon-viewport: 6.1.10 => 6.1.10
    @storybook/addons: 6.1.10 => 6.1.10
    @storybook/source-loader: 6.1.10 => 6.1.10
    @storybook/vue: 6.1.10 => 6.1.10
```

After upgrading to latest:
```
Environment Info:

  System:
    OS: Linux 4.4 Ubuntu 20.04.1 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
  Binaries:
    Node: 12.19.0 - ~/.nvm/versions/node/v12.19.0/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v12.19.0/bin/npm
  npmPackages:
    @storybook/addon-a11y: 6.1.14 => 6.1.14
    @storybook/addon-actions: 6.1.14 => 6.1.14
    @storybook/addon-backgrounds: 6.1.14 => 6.1.14
    @storybook/addon-centered: 5.3.21 => 5.3.21
    @storybook/addon-contexts: 5.3.21 => 5.3.21
    @storybook/addon-docs: ^6.1.14 => 6.1.14
    @storybook/addon-essentials: ^6.1.14 => 6.1.14
    @storybook/addon-knobs: 6.1.14 => 6.1.14
    @storybook/addon-links: 6.1.14 => 6.1.14
    @storybook/addon-notes: 5.3.21 => 5.3.21
    @storybook/addon-options: 5.3.21 => 5.3.21
    @storybook/addon-storyshots: 6.1.14 => 6.1.14
    @storybook/addon-storysource: 6.1.14 => 6.1.14
    @storybook/addon-viewport: 6.1.14 => 6.1.14
    @storybook/addons: 6.1.14 => 6.1.14
    @storybook/source-loader: 6.1.14 => 6.1.14
    @storybook/vue: 6.1.14 => 6.1.14
```

**Additional context**

**Comment by fbtwitter** [comment]
It happens to me in nuxt.. where those storysource won't generate.. after adding some parameter data like background, action, etc

**Comment by martell2010** [comment]
I have the same problem, but steps to reproduce other.
I have a table in component template and I call method inside. When this method return value story is work. When method return other sync method I got this error.

Worked
```js
<template>
  <div class="hello">
    <table>
      <tr>
        <th>col 1</th>
        <th>col 2</th>
        <th>col 2</th>
      </tr>
      <tr v-for="i in 10" :key="i">
        <td>{{ i }}</td>
        <td>column 2.{{ i }}</td>
        <td>{{ getValue(i) }}</td>
      </tr>
    </table>
  </div>
</template>

<script>
export default {
  name: "WorkedTable",
  methods: {
    getValue(i) {
       return `MDX storie is worked ${i}`;
    }
  },
};
</script>
```
Broken
```js
<template>
  <div class="hello">
    <table>
      <tr>
        <th>col 1</th>
        <th>col 2</th>
        <th>col 2</th>
      </tr>
      <tr v-for="i in 10" :key="i">
        <td>{{ i }}</td>
        <td>column 2.{{ i }}</td>
        <td>{{ getValue(i) }}</td>
      </tr>
    </table>
  </div>
</template>

<script>
export default {
  name: "BrokenTable",
  methods: {
    brokenFn(i) {
      return `MDX storie is broken ${i}`;
    },
    getValue(i) {
      return this.brokenFn(i);
    },
  },
};
</script>
```

**Comment by vanessayuenn** [comment]
Hi! I am going through and cleaning up old issues in the repo. We just shipped a ton of improvements in Storybook 7.0, and I suspect this bug might be fixed already. Can you please try it out? If this issue persists with the latest version, please open a new bug report with a 7.x reproduction. Thank you so much! 🙏🏼
